runtime/Java/src/org/antlr/v4/runtime/VocabularyImpl.java

```diff
@@ -94,6 +96,8 @@ public VocabularyImpl(String[] literalNames, String[] symbolicNames, String[] di
 		this.literalNames = literalNames != null ? literalNames : EMPTY_NAMES;
 		this.symbolicNames = symbolicNames != null ? symbolicNames : EMPTY_NAMES;
 		this.displayNames = displayNames != null ? displayNames : EMPTY_NAMES;
+		this.maxTokenType = Math.max(this.displayNames.length,
```

Comment:
This is very conservative as it even include display names. Pinging @sharwell for feedback on this one.